src/packages/arithmetic.rs

```diff
@@ -362,7 +362,13 @@ def_package!(crate:ArithmeticPackage:"Basic arithmetic", lib, {
     #[cfg(feature = "decimal")]
     {
         lib.set_fn_1("sign", |x: Decimal| {
-            Ok(if x.is_sign_positive() { 1 } else { -1})
+            Ok(if x == Decimal::zero() {
+                0
```

Comment:
You have to be careful about types here.    This may not be `INT`.

You'll need to do `0 as INT` etc.